app/controllers/bookings_controller.rb

```diff
@@ -93,25 +79,13 @@ def booking_params
     )
   end
 
-  def parse_query_string(parameters)
-    Rack::Utils.parse_nested_query(parameters)
+  def services(booking = nil)
+    BookingsServices.new(booking)
   end
 
-  def parse_params(obj)
-    result = {}
-    obj.map do |k, v|
-      result[k] = v
-    end
-    result
-  end
-
-  def send_booking_email(booking, user)
-    if logged_in?
-      BookingMailer.successful_booking(booking, user).deliver_later
-    else
-      booking.passengers.map do |passenger|
-        BookingMailer.successful_booking(booking, passenger).deliver_later
-      end
-    end
+  def successful_booking(booking, msg)
+    services(booking).booking_email(current_user)
+    flash[:success] = "Booking successfully #{msg}."
+    redirect_to booking
   end
 end
```

Comment:
**Code quality - Critical**

Slim the `BookingsController`. 
Move some of the methods to a newly introduced `BookingsService` service object.